packages/runtime-common/indexer.ts

```diff
@@ -174,6 +169,7 @@ export class Indexer {
       'AND',
       ...realmVersionExpression({ useWorkInProgressIndex: true }),
       'ORDER BY i.card_url COLLATE "POSIX"',
+      ...(page ? [`LIMIT ${page.size} OFFSET ${page.number * page.size}`] : []),
```

Comment:
so the idea here is that we first figure out how many invalidations there are (so a `COUNT` query). Then we make multiple requests for each page of invalidations to gather them all up. The key here is that we need to hold the realm version constant so that if there are no changes to the results between the pages.

In order to hold the realm version constant you will need to pass in the realmversion that you want to use into this function, and then use `...realmVersionExpression({ withMaxVersion: versionNumber} )` to lock the request to a particular version number. the version number that you'll want to pass in will be the `Batch.realmVersion`--that is the WIP index version.

Reply:
Another idea is to not both with the COUNT and just keep running the query one page at a time until you get no results. that's probably cleaner...

Reply:
I think a reasonable page size is probably 1000 items. that can be hard coded